CMakeLists.txt

```diff
@@ -1,19 +1,23 @@
 cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
 
+project(Reaping2)
+
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
 set( CMAKE_CXX_STANDARD 11 )
 
 # support for YouCompleteMe
 set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
-if(UNIX)
+
+if( CMAKE_HOST_UNIX )
     set( CMAKE_CXX_FLAGS "-Wno-deprecated-declarations" )
-endif(UNIX)
-if($ENV{REAPING2_PROFILE})
-    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pg" )
-endif($ENV{REAPING2_PROFILE})
+    if($ENV{REAPING2_PROFILE})
+        set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pg" )
+    endif($ENV{REAPING2_PROFILE})
 
-project(Reaping2)
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -g -Wall")
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -g -Wall")
```

Comment:
During release build, CMake already appends `-O3`, so it seems redundant.
Also, instead of adding `-g` to the release flags, how about using `-DCMAKE_BUILD_TYPE=RelWithDebInfo` ? By default, that configuration type adds: `-O2 -g -DNDEBUG`.